tree.go

```diff
@@ -12,8 +12,9 @@ import (
 
 // Tree is an immutable AVL+ Tree. Note that this tree is not thread-safe.
```

Comment:
Tree is actually a reference to a the root of an immutibel AVL+ Tree
Plus mutable metadata.

Make it clear that root is immutiable, but tree is mutable